content/post/70-js-ipfs-0-38.md

```diff
@@ -0,0 +1,129 @@
+---
+date: 2019-08-30
+url: 70-js-ipfs-0-38
+title: js-ipfs 0.38.0 released
+author: Alex Potsides
+---
+
+> Garbage collection, gossipsub, IPNS support for ipfs resolve, oh my!
+
+```console
+$ npm install -g ipfs
+```
+
+# 🔦 Highlights
+
+> Garbage collection, gossipsub & IPNS support for `ipfs resolve`! Also a new release process!
+
+## 🗑️ Garbage collection
+
+Support for garbage collection has shipped with `0.38.0`! This means that blocks in your IPFS Repo will be removed whenever gc is run unless they are pinned.
+
+This is essential for keeping your repo to a manageable size, but must be run manually for the time being.
+
+Use it with the following command:
+
+```console
+$ jsipfs repo gc
+removed Qmfoo
+removed Qmbar
+```
+
+To prevent blocks being collected, pin them:
+
+```console
+$ jsipfs pin add Qmbaz
+pinned Qmbaz recursively
+```
+
+## 💬 Gossipsub enabled by default
+
+Pubsub is now enabled by default with Gossipsub as the default implementation.
+
+More efficient than [Floodsub](https://github.com/libp2p/js-libp2p-floodsub), [Gossipsub](https://github.com/ChainSafe/gossipsub-js) creates an overlay on top of the network of nodes which propagates messages to a subset of subscribers instead of broadcasting messages to everyone.
+
+Read more about the design at [`libp2p/specs/pubsub/gossipsub`](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub)
+
+## 📛 IPNS support for `ipfs resolve`
+
+You can now resolve [IPNS](https://docs.ipfs.io/guides/concepts/ipns/) names via `ipfs resolve`, what's more the recursive option is now on by default.
+
+```console
+$ jsipfs resolve /ipns/Qmqux
+/ipfs/Qmgarply
+```
+
+## 🎁 New relase process & a call for early testers
+
+This is the first release to follow our new, more predicatable, more rigorous release process. It's outlined in the [release issue template](https://github.com/ipfs/js-ipfs/blob/master/doc/RELEASE_ISSUE_TEMPLATE.md), but in short we have defined steps, with an addional period for internal & external testing and gathering feedback on the changes.
+
+It's also the first release to ship with lockfiles for `yarn` and `npm`. This is to ensure that you only get the dependcies that we have validated with CI, and not untested code that may have been released with breaking changes since CI last ran on the release candidate.
```

Comment:
```suggestion
It's also the first release to ship with lockfiles for `yarn` and `npm`. This is to ensure that you only get the dependencies that we have validated with CI, and not untested code that may have been released with breaking changes since CI last ran on the release candidate.
```